clients/edgex-foundry/device_client.go

```diff
@@ -0,0 +1,315 @@
+package edgex_foudry
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+	"github.com/go-logr/logr"
+	"github.com/go-resty/resty/v2"
+	devicev1alpha1 "github.com/openyurtio/device-controller/api/v1alpha1"
+	edgeCli "github.com/openyurtio/device-controller/clients"
+)
+
+type ClientURL struct {
+	Host string
+	Port int
+}
+
+type EdgexDeviceClient struct {
+	*resty.Client
+	CoreMetaClient    ClientURL
+	CoreCommandClient ClientURL
+	logr.Logger
+}
+
+func NewEdgexDeviceClient(coreMetaClient, coreCommandClient ClientURL, log logr.Logger) *EdgexDeviceClient {
```

Comment:
why use `logr.Logger` instead of `k8s.io/klog`?

Reply:
Thanks for your suggestion, and I will submit a PR version to replace all `logr.Logger` with `k8s.io/klog`